### PR TITLE
Updates add-flutter-fragment.md instructions to latest embedding API.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Updates add-flutter-fragment.md instructions to latest embedding API.

### CODE REVIEW

File: src/docs/development/add-to-app/android/add-flutter-fragment.md [hunk 1]
```diff
@@ -157,29 +164,31 @@ Please see the [instructions for instantiating and starting a `FlutterEngine`].
...
+  .getInstance()
+  .put("my_engine_id", flutterEngine);
+
+//....sometime later...
+
+// In your Activity.
+flutterFragment.withCachedEngine("my_engine_id").build();
 ```
+`FlutterFragment` internally knows about `FlutterEngineCache` and retrieves the
+pre-wared `FlutterEngine` based on the ID given to `withCachedEngine()`.
```
Comment (sfshaza2) [inline src/docs/development/add-to-app/android/add-flutter-fragment.md]: pre-wared => pre-warmed

File: src/docs/development/add-to-app/android/add-flutter-fragment.md [hunk 2]
```diff
@@ -203,12 +212,13 @@ initial routes (routes other than `/`). To facilitate this, `FlutterFragment`'s
...
 ```java
-FlutterFragment flutterFragment = new FlutterFragment.Builder()
+// With a new FlutterEngine.
+FlutterFragment flutterFragment = FlutterFragment.withNewEngine()
     .initialRoute("myInitialRoute/")
     .build();
 ```
 
-{{site.alert.note}}
+{{site.alert.note}}
```
Comment (sfshaza2) [inline src/docs/development/add-to-app/android/add-flutter-fragment.md]: I think it's amusing that SO many PRs I see remove trailing white space, but your PRs ADD it. I always figured it was a side effect of whatever editing tool is being used.

File: src/docs/development/add-to-app/android/add-flutter-fragment.md [hunk 3]
```diff
@@ -308,38 +330,46 @@ Some apps choose to use `Fragment`s as entire Android screens. In these apps, it
...
+// Using a cached FlutterEngine.
+FlutterFragment flutterFragment = FlutterFragment.withCachedEngine("my_engine_id")
     .shouldAttachEngineToActivity(false)
     .build();
 ```
 
 Passing `false` to the `shouldAttachEngineToActivity()` `Builder` method
 prevents Flutter from interacting with the surrounding `Activity`. The default
-value is `true`, which allows Flutter and Flutter plugins to interact with
+value is `true`, which allows Flutter and Flutter plugins to interact with the
```
Comment (sfshaza2) [inline src/docs/development/add-to-app/android/add-flutter-fragment.md]: Good catch.

File: src/docs/development/add-to-app/android/add-flutter-fragment.md (hunk 1 above)
Comment (matthew-carroll) [reply src/docs/development/add-to-app/android/add-flutter-fragment.md]: Done

File: src/docs/development/add-to-app/android/add-flutter-fragment.md (hunk 2 above)
Comment (matthew-carroll) [reply src/docs/development/add-to-app/android/add-flutter-fragment.md]: That is weird. I tried to remove it, hopefully it worked.

File: src/docs/development/add-to-app/android/add-flutter-fragment.md [hunk 4]
```diff
@@ -308,38 +330,46 @@ Some apps choose to use `Fragment`s as entire Android screens. In these apps, it
 would be reasonable for a `Fragment` to control system chrome like Android's
 status bar, navigation bar, and orientation.
 
-<!-- TODO(mattcarroll): to add back. -->
-<!-- add-flutter-fragment_fullscreen was here-->
+{% asset
+development/add-to-app/android/add-flutter-fragment/add-flutter-fragment_fullscreen.png
```
Comment (xster) [inline src/docs/development/add-to-app/android/add-flutter-fragment.md]: @matthew-carroll it's this PR. I'm going to comment these out for now so I can keep building. Don't forget to put this back along with their assets later.

Comment (matthew-carroll) [reply src/docs/development/add-to-app/android/add-flutter-fragment.md]: I have a PR up with the assets. Can you approve that PR and leave these tags as-is?

Comment (xster) [reply src/docs/development/add-to-app/android/add-flutter-fragment.md]: SG thanks

**Pull request merged**